truss-chains/truss_chains/framework.py

```diff
@@ -1211,133 +1212,195 @@ def decorator(cls: Type[ChainletT]) -> Type[ChainletT]:
     return decorator(cls_or_chain_name)  # Decorator used without arguments
 
 
-def _get_entrypoint_chainlets(symbols) -> set[Type[definitions.ABCChainlet]]:
-    return {
-        sym
-        for sym in symbols
-        if utils.issubclass_safe(sym, definitions.ABCChainlet)
-        and cast(definitions.ABCChainlet, sym).meta_data.is_entrypoint
-    }
+class _ABCImporter(abc.ABC):
+    @classmethod
+    @abc.abstractmethod
+    def no_entrypoint_error(self, module_path: pathlib.Path) -> ValueError:
+        pass
 
+    @classmethod
+    @abc.abstractmethod
+    def multiple_entrypoints_error(
+        self, module_path: pathlib.Path, entrypoints: set[type[definitions.ABCChainlet]]
+    ) -> ValueError:
+        pass
 
-@contextlib.contextmanager
-def import_target(
-    module_path: pathlib.Path, target_name: Optional[str] = None
-) -> Iterator[Type[definitions.ABCChainlet]]:
-    resolved_module_path = pathlib.Path(module_path).resolve()
-    modules_before = set(sys.modules.keys())
-    module, loader = _load_module(module_path)
-    modules_after = set()
-
-    chainlets_before = _global_chainlet_registry.get_chainlet_names()
-    chainlets_after = set()
-    try:
+    @classmethod
+    @abc.abstractmethod
+    def target_cls_type(self) -> Type[definitions.ABCChainlet]:
+        pass
+
+    @classmethod
+    def _get_entrypoint_chainlets(self, symbols) -> set[Type[definitions.ABCChainlet]]:
+        return {
+            sym
+            for sym in symbols
+            if utils.issubclass_safe(sym, self.target_cls_type())
+            and cast(definitions.ABCChainlet, sym).meta_data.is_entrypoint
+        }
+
+    @classmethod
+    def _load_module(
```

Comment:
`_load_module` and `_cleanup_module_imports` have no changes, other than the ones needed to make them class methods